src/subcommand.rs

```diff
@@ -255,10 +255,22 @@ impl Subcommand {
     Ok(())
   }
 
-  fn format(config: &Config, ast: Ast, search: &Search) -> Result<(), Error<'static>> {
+  fn format(config: &Config, ast: Ast, src: &str, search: &Search) -> Result<(), Error<'static>> {
     config.require_unstable("The `--fmt` command is currently unstable.")?;
 
-    if let Err(io_error) =
+    if config.dry_run {
+      let new_src = ast.to_string();
+      if src != new_src {
+        use prettydiff::diff_words;
+        println!("Formatting diff:\n\n{}", diff_words(&src, &new_src));
+        Err(Error::FormatCheckFoundDiff)
+      } else {
+        if config.verbosity.loud() {
+          eprintln!("No format difference found.");
```

Comment:
I think it's actually okay to not print anything in case of success. It's very common to use `rust fmt -- --check` and 
only check for the exit code.

```suggestion
```